### PR TITLE
Set of docs for Apps

### PR DESCRIPTION
**Pull request by lkostrowski**
This is trunk branch for docs, more PRs will be opened to this branch

### CODE REVIEW

File: docs/developer/extending/apps/architecture/requirements/required-endpoints.mdx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+See [Manifest docs](https://docs.saleor.io/docs/3.x/developer/extending/apps/manifest) or check its [typings](https://github.com/saleor/saleor-app-sdk/blob/main/src/types.ts#L67).
+
+If your app is built with [Next.js](https://nextjs.org/), you can use a [helper from saleor/app-sdk](https://github.com/saleor/saleor-app-sdk/blob/main/docs/api-handlers.md#manifest-handler-factory), that creates a handler with minimal config.
+
+Dashboard and Saleor API will use this endpoint to install the app.
+
+# Register URL
+
+- Method: `POST`
+- Response - status `200`
```
Comment (lkostrowski) [inline docs/developer/extending/apps/architecture/requirements/required-endpoints.mdx:28]: @krzysztofwolski SDK returns also {success: true}, is this checked by dashboard/saleor?

Comment (krzysztofwolski) [reply docs/developer/extending/apps/architecture/requirements/required-endpoints.mdx:28]: https://docs.saleor.io/docs/3.x/developer/extending/apps/installing-apps 
According to docs, only the status is required. The success message seems to be an old convention.

To be sure checked registration using handler:
```typescript
import { NextApiHandler, NextApiResponse } from "next";

export const handler: NextApiHandler = (req, res) => {
  res.status(200);
  return;
};

export default handler;

```

And API marked app as registered

**Comment by lkostrowski** [comment]
@peelar I think it can be reviewed. There are few missing pages but we can hide them or even leave as they are now

**Comment by peelar** [comment]
> @peelar I think it can be reviewed. There are few missing pages but we can hide them or even leave as they are now

I will push my changes straight away. We can still discuss them (each page is a separate commit) through the GH review.

**Comment by peelar** [comment]
@krzysztofwolski @lkostrowski I am still in the process of reviewing it, although I am close to the end. In the meantime, I would like to propose a new structure for the "Apps" docs. Once agreed, I can introduce it. Here it is:

### The current structure:
```
Extending Saleor
- Apps
	- The App Manifest
	- Installing and Managing Apps
	- Asynchronous Webhooks
	- Synchronous webhooks
		- ...
	- Subscription Webhook payloads
	- Extending Dashboard with Apps
	- Webhook payloads
	- Tunneling Apps
	- Architecture
		- ...
	- Developing Apps
		- ...
	- Deploying Apps
		- ...
```

### The proposed structure:
```
Extending Saleor
- Apps
	- Architecture
		- ...Architecture (the current content of "Architecture")
		- The App Manifest
	- Webhooks
		- Asynchronous Webhooks
		- Synchronous webhooks
			- ...
		- Subscription Webhook payloads
		- Webhook payloads
	- Developing Apps
		- Installing and Managing Apps
		- Tunneling Apps
		- Extending Dashboard
		- ...
	- Deploying Apps
		- ...
```

The goals behind it were:
- reduce the cognitive overload when opening the "Apps" subcategory
- group some topics together (like webhooks)

**Comment by peelar** [comment]
> Looks good, we should probably have some introduction page with navigation too.

@lkostrowski I will add some content to the current ["Apps" introductory page](https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts).

> Maybe we can even extract "apps" outside of "extending saleor"?

That would require some bigger structure overhaul but I am planning to do that in the next few days. I will keep that in mind.

**Comment by peelar** [comment]
@lkostrowski

A revamped ["Apps" introductory page](https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts):
image

**Comment by peelar** [comment]
@lkostrowski @krzysztofwolski 

Here is the proposed structure. I already pushed changes to the branch:
Closed:
Capture-2022-11-17-170406

Open:
Capture-2022-11-17-170430

**Comment by lkostrowski** [comment]
@peelar looks good :) What else do we need to merge?

**Comment by peelar** [comment]
> @peelar looks good :) What else do we need to merge?

Nothing from my side

**Pull request merged**